app/services/xccdf_report_parser.rb

```diff
@@ -31,6 +32,7 @@ def inventory_api
 
   def save_host
     @host = Host.find_or_initialize_by(
+      id: @metadata&.dig('insights_id'),
```

Comment:
I wonder what should we do in the case the insights_id or the metadata are not available. Perhaps we should just refuse to use the message and fail the validation?

Reply:
A few lines below, we call `inventory_api.sync`. Is that necessary at all at this point? We are already taking in the insights_id coming from the inventory

Reply:
I think we can fail the validation, but I'm still a bit worried the metadata isn't making it to the parser in prod or CI. We need to double check before we push this and potentially break all the compliance uploads.

What should we do if `@metadata['fqdn']` doesn't equal `report_xml.search('target').text` ? Should we even check?

Reply:
What if the host doesn't exist in the inventory? We need to verify that it is created without the `inventory_api.sync` call.